internal/engine/wazevo/backend/isa/amd64/lower_mem.go

```diff
@@ -1,10 +1,159 @@
 package amd64
 
 import (
+	"github.com/tetratelabs/wazero/internal/engine/wazevo/backend"
+	"github.com/tetratelabs/wazero/internal/engine/wazevo/backend/regalloc"
 	"github.com/tetratelabs/wazero/internal/engine/wazevo/ssa"
 )
 
+var addendsMatchOpcodes = [5]ssa.Opcode{ssa.OpcodeUExtend, ssa.OpcodeSExtend, ssa.OpcodeIadd, ssa.OpcodeIconst, ssa.OpcodeIshl}
+
+type (
+	addend64 struct {
+		r     regalloc.VReg
+		off   int64
+		shift byte
+	}
+)
```

Comment:
64 seems unnecessary


```suggestion
type addend struct {
	r     regalloc.VReg
	off   int64
	shift byte
}
```